added 'unload_zbarcam' that unload zbarcam.kv and xcamera.kv files

qlrd opened the pull request:
This PR add a `unload_zbarcam` method.

This method fixes the issue "[restart after releasing the camera results in these warnings followed by pages of errors](https://github.com/kivy-garden/zbarcam/issues/59#issuecomment-1280102277)"

The solution was to unload `zbarcam.kv` and `xcamera.kv` files (see this [response](https://github.com/kivy-garden/zbarcam/issues/59#issuecomment-1784158670)).

A working example with this approach can be found [here](https://github.com/qlrd/krux-file-signer/tree/kivy).

qlrd committed:
added 'unload_zbarcam' that unload zbarcam.kv and xcamera.kv files

AndreMiras commented on src/kivy_garden/zbarcam/zbarcam.py:
```diff
@@ -210,3 +211,19 @@ def stop(self):
             self.xcamera._camera._release_camera()
+
+    def unload_zbarcam(self):
+        id ZBarCam.kv_loaded:
```
Did you mean `if`?
Pity we no longer have working tests for this repo, this would have been caught easily

qlrd replied:
Yes, sorry for typo.

I will fix this and create a new commit

qlrd replied:
> Pity we no longer have working tests for this repo, this would have been caught easily

Maybe I can work with some github-action to do this.

AndreMiras replied:
Yeah we would love to :+1:

qlrd committed:
Update zbarcam.py

Fixed typo: `id` to `if` on `unload_zbarcam method

AndreMiras reviewed (commented):
Thanks for the update.
I tried to find where that approach is used in the code you linked, but I didn't find it as in [b68fe7b](https://github.com/qlrd/krux-file-signer/tree/b68fe7b) I GitHub searched for `unload_file`, but got no results.
I would it make sense to call this method as we call the stop method?

qlrd commented:
> I tried to find where that approach is used in the code you linked

Sorry again 😭 

Didn't add this as `unload_zbarcam` or `unload_file`. I only [hardcoded it](https://github.com/qlrd/krux-file-signer/blob/kivy/src/screens/scan.py#L127).

> I would it make sense to call this method as we call the stop method?

I think it make sense, since the `stop` method dint unload `.kv` files and if someone want to reload zbarcam, the App can crash (like what happened with me). I would like to point 3 things:

1. Maybe call `unload_files` since it unload either `zbarcam.kv` and `xcamera.kv`? Or will this bring more confusion? (maybe call it simply `unload`?)
2. Maybe make it as a "protected" method `_unload` and call it on `stop` method?
3. I did not tested on android; maybe there might be some strange behavior for smartphones using this?

I will wait for response before make any new commit

AndreMiras commented:
Yes for 1 & 2 `_unload()`, let's go for it.
Good to know that you didn't get a chance to test on Android. I'd like to make sure it doesn't introduce regressions on Android indeed.

qlrd committed:
updated

added release device step for non android devices in  method

changed  method to

qlrd commented:
> Yes for 1 & 2 `_unload()`, let's go for it. Good to know that you didn't get a chance to test on Android. I'd like to make sure it doesn't introduce regressions on Android indeed.

Done.

I added a step for releasing devices on non android devices in `stop` method.

Sorry for "incomplete" commit message. It was suposed to be:

> updated `zbarcam.py`
added release device step for non android devices in `stop` method
changed `_unload_zbarcam` method to `_unload`

qlrd commented:
@AndreMiras, did you tested the code?

AndreMiras commented:
I didn't get a chance to. Do you mind giving it a try?

qlrd commented:
> I didn't get a chance to. Do you mind giving it a try?

I can try, but i need some time to apply
